### DISCUSSION TITLE
OMEdit - autocompletion doesn't work well

### DISCUSSION THREAD

**Discussion by Czesiek1701** [OpenModelica/OpenModelica#12101] [Q&A]
## Description

Hi!
After few reinstallations OMEdit autocompletion during coding stops working.
When I refer to a class or functions after dot typing program is suggesting me next classes or attributes etc. But when I have object of other class or just variable declared by me the auto-completion is not working. 

model Example
Modelica.Blocks.Interfaces.BooleanInput myInput; // <---  suggested inside classes all time
Real someVar;  // <--- suggested Real
equation
m... ;  // <--- suggest only model, Modelica, ModelicaServices without declared myInput
s...; // <--- nothing
end Example;

![image](https://github.com/OpenModelica/OpenModelica/assets/157902583/b809e37e-902e-4591-9b1f-daf1cc103633)

After save also doesn't work. 

## Steps to reproduce

With imported libraries in 4.0.0 version: _OpenModelica_, _ModelicaServices_, _Complex_, _Modelica_.
New model with name: _Auto\_completion\_test_

`model Auto_completion_test`
`Real myReal;`
`Modelica.Blocks.Continuous.Integrator myIntegrator;`
`equation`
`end Auto_completion_test;`

Save. Close OM. Run OM. Open model.

## Expected Behaviour

When typing in equation: 

'...'
`equation`
`my`   <--- It should suggest myReal and myIntegrator to chose by tab
`end Auto_completion_test;`

## Version and OS

OpenModelica v1.22.2 (64-bit)
OMSimulator v2.1.1.post247-g614c663-mingw-notlm-debug
OS Windows 11 Home 64-bit 
OM Libraries version 4.0.0

Since 3 days of coding I haven't seen any suggestion by OM for any component in any model.

Thank you for any help!

**Comment by casella** [comment]
@czesiek1701 autocompletion shows you the classes or components that are loaded in the OMC instance running in the background. While you are typing code, the OMC in the background is not continuously updated, mainly because the code is most of the time invalid during editing. It is updated when you save the code and when you switch to another windows. Please try that.

After saving it works for me, maybe you can post a specific example. Please also specify which version of OMEdit you are using.

**Comment by Czesiek1701** [comment]
@casella Thank you for response! No, it's still not working. I am creating a model, it has some new variables, compiled and simulated muny times. OM also was few times closed and opened. This is not working also in other old models and copied from Standard Library. I was coding during half year and never see something like that. Only declared objects are not being sugested. I must delete sth by accident or change some path.

**Reply by casella** [reply]
The only way we can help you is if you give us an example to replicate the problem and explain exactly how to replicate it in a ticket. Thanks!

**Reply by Czesiek1701** [reply]
I upadated post with steps to reproduce and versions. I think that issue can exists only on my computer.
Thank you

**Reply by AnHeuermann** [reply]
This is one of the things I want for the Modelica language server and it's not too far off. We already know all of the definitions in a single file. Respecting the scope and registering the handle isn't done yet:
https://github.com/OpenModelica/modelica-language-server/issues/14
Oh and working with multiple files is quiet the task, see https://github.com/OpenModelica/modelica-language-server/issues/7

**Comment by adeas31** [comment]
The autocompletion is not adapted to the new instance API.
As a workaround disable the use of new instance API in settings `Tools->Options->Disable new instance-based graphical editing of models`.

**Reply by Czesiek1701** [reply]
It works! Thank you very much!

**Comment by Czesiek1701** [comment]
Thank you guys for help! Problem solved :)

**Comment by adeas31** [comment]
We recommend to use the instance API. With PR #12104 I fixed the auto-complete for instance API.
Test it with the nightly build tomorrow.
